Automattic/wp-calypso
# Invites: Fix Invite Accept Form Title

Fixes #95171

## Proposed Changes

There are two variants of the Invite Accept form that are relevant here: `InviteAcceptLoggedIn` and `InviteAcceptLoggedOut`. We also have `InviteFormHeader` which, as the name applies, is basically a heading for both those screens. But at the moment, something strange happens: `InviteFormHeader` is **only** used in `InviteAcceptLoggedIn`. In `InviteAcceptLoggedOut`, the code is extracted from `InviteFormHeader` and used manually (but not all of the code is, which is what causes the current bug).

To handle this better, I see two possible solutions:

1. Make `InviteAcceptLoggedIn` also use the extracted functions
2. Just use `InviteFormHeader` directly in both.

This PR does the second because I think it makes more sense to treat the form header like a component that can be used in both screens. For that reason, I've done the following:

* Moved `getExplanationForInvite` into a central location rather than having that function being called individually (but weirdly, only in the logged out screen and not the logged in screen)
* Treated it as a component, so that even though the styles are different for the logged in/out screen, they're included within the parent stylesheet instead.  

## Why are these changes being made?

* Currently when you invite a reader to a private site, the Invite Form title says "Sign up to start editing". This fixes that text to be more accurate since viewers cannot edit sites. 

## Testing Instructions

1. Create a private site on Account A
2. Invite Account B to be a viewer of that site (make sure you haven't actually registered Account B yet!)
3. Accept the invite link 
4. Confirm that you're now prompted to view the site rather than edit it (the strings were already there, and in the interests of i18n, I haven't changed them)

| Before | After |
|--------|--------|
| <img width="613" alt="Screenshot 2024-10-05 at 13 32 44" src="https://github.com/user-attachments/assets/713c23e5-9f57-4921-bfa8-469c74bf7051"> | <img width="745" alt="Screenshot 2024-10-05 at 13 32 31" src="https://github.com/user-attachments/assets/9f584e25-d5c3-4fe3-92bc-0a7c6b8ab99a"> |

5. Just in the interests of completeness, it might make sense to check the other use of `InviteFormHeader`. This is basically the same steps, but sending the invite to an email already with a WP.com account which you're signed into.

| Before | After |
|--------|--------|
| <img width="482" alt="Screenshot 2024-10-05 at 14 51 41" src="https://github.com/user-attachments/assets/cba602ea-364b-436e-8538-963721a1f00c"> | <img width="469" alt="Screenshot 2024-10-05 at 14 51 37" src="https://github.com/user-attachments/assets/c2c9e989-70eb-4018-8a4d-0b9ea9323452"> | 

I removed some styles that are inherited with this change, but there should be no real difference. I might be misunderstanding what this code is trying to do, but I hope these changes make sense!

cc @jsnajdr, @renancarvalho, @agrullon95
